### PR TITLE
feat: Allow `token` to be `null` in `usePlaidLink` hook

### PR DESCRIPTION
**Pull request by skylarmb**
Testing out some solutions proposed to solve #128 

This PR allows `token` to be `null` in the initial config for the `usePlaidLink` hook. This allows developers to greatly simplify their code by generating a link token asynchronously within their Plaid Link component instead of requiring a parent and child component to manage the link token.

This also allows developers to more easily preload the Link iframe in the background as it allows initializing the Link javascript on the page without generating a link token (which is possible in vanilla js, but previously not possible with React).

Previous behavior: If `token` is null when passed to `usePlaidLink`, an uncaught exception is thrown. There is no way for developers to catch this error other than with a React `ErrorBoundary` 

New behavior: If `token` is null, the link js/iframe still is still injected into the page but `ready` remains `false` until a valid `token` is provided

This change has already been published as `v3.3.0-beta.0` and has been tested by developers: https://github.com/plaid/react-plaid-link/issues/128#issuecomment-961518620

### CODE REVIEW

**Comment by skylarmb** [comment]
Discussions around pros and cons:

Pros:
- Makes integrating with this SDK easier by eliminating the need for a wrapper component
- Makes it possible to preload Link iframe without a token

Cons:
- Creates more potential for people to shoot themselves in the foot, e.g. no error thrown if config is invalid
- Typescript type becomes more flexible meaning its possible for devs to accidentally leave out `token` entirely and expect Link to work based on the types

File: src/types/index.ts [hunk 1]
```diff
@@ -139,7 +139,7 @@ export type PlaidLinkOptionsWithPublicKey = CommonPlaidLinkOptions & {
 export type PlaidLinkOptionsWithLinkToken = CommonPlaidLinkOptions & {
   // Provide a link_token associated with your account. Create one
   // using the /link/token/create endpoint.
-  token: string;
+  token: string | null;
```
Comment (skylarmb) [inline src/types/index.ts:142]: this is the key change

File: src/usePlaidLink.ts [hunk 2]
```diff
@@ -21,19 +21,30 @@ const noop = () => {};
...
   useEffect(() => {
     // If the link.js script is still loading, return prematurely
     if (loading) {
       return;
     }
 
+    // If the token is undefined, return prematurely
+    if (!options.token) {
+      return;
+    }
```
Comment (skylarmb) [inline src/usePlaidLink.ts:46]: this is the key change, as discussed in #128

**Review by phoenixy1 — DISMISSED** [review]
can i stamp this? let's find out!

**Review by phoenixy1 — APPROVED** [review]

**Review by phoenixy1 — APPROVED** [review]

**Pull request merged**